### ISSUE TITLE
there are some proplems with fio_str_tmp function

### ISSUE THREAD

**Issue by immortalChensm** [boazsegev/facil.io#82]
this is my source code   
```c  
typedef struct table_data{
    char **fields;
    char **value;
}table_data;
char *fields[] = {"id","name","age"};
    char *value[] = {"1","jack","100"};
    insert("users",(table_data){.fields=fields,.value=value}); 
int insert(char *table,table_data data)
{
    int i=0;
    FIOBJ fields = fiobj_str_tmp();
    FIOBJ val = fiobj_str_tmp();
    if (table){
        while (data.fields[i]){
            fiobj_str_write(fields,"`",1);
            fiobj_str_write(fields,data.fields[i],strlen(data.fields[i]));
            fiobj_str_write(fields,"`",1);
            fiobj_str_write(fields,",",1);

            fiobj_str_write(val,"'",1);
            fiobj_str_write(val,data.value[i],strlen(data.value[i]));
            fiobj_str_write(val,"'",1);
            fiobj_str_write(val,",",1);
            i++;
        }
        printf("fields=%s\n",fiobj_obj2cstr(fields).data);
        printf("value=%s\n",fiobj_obj2cstr(val).data);

    }
}
```   
run it shows 
```c 
fields=`id`,'1',`name`,'jack',`age`,'100',
value=`id`,'1',`name`,'jack',`age`,'100',

```

**Comment by boazsegev** [comment]
Hi @immortalChensm ,

Thank you for opening this issue.

This is a design issue, solved in the 0.8.x branch, where the `fiobj_str_tmp` _function_ is deprecated in favor of a **macro** based solution that allows multiple temporary strings to be placed on the stack.

Under the current design, the `fiobj_str_tmp` returns a thread-safe static variable. It is limited to a single temporary string per thread.

As it stands, there is no way to fix the `fiobj_str_tmp` function.

I agree with you that this design is suboptimal and so, it was deprecated.

However, allocating a string with `fiobj_str_new` should be fast enough... even if you need to  remember to free the string with `fiobj_free`.

I hope this helps.

Kindly,
   Bo.

**Comment by boazsegev** [comment]
P.S. - I closed this issue because it's solved in 0.8.0 (pre-beta).

Thank you again for opening the issue.
   Bo.

**Comment by immortalChensm** [comment]
Thank you very much for your reply. I see
